impeller/renderer/backend/vulkan/swapchain_impl_vk.cc

```diff
@@ -469,11 +490,12 @@ bool SwapchainImplVK::Present(const std::shared_ptr<SwapchainImageVK>& image,
             // Vulkan guarantees that the set of queue operations will still
             // complete successfully.
             [[fallthrough]];
-          case vk::Result::eSuccess:
-            is_rotated_ = false;
-            return;
           case vk::Result::eSuboptimalKHR:
-            is_rotated_ = true;
+            // In theory we shouldn't receive this (because we're polling above)
+            // but we shouldn't handle it as a validation error if we do receive
+            // it.
```

Comment:
I think this isn't correct - right. Won't we be presenting with suboptimalkhr on every frame when we are rotated? (give that we aren't doing vulkan prerotation). 

Don't try this on the S10 😄

Reply:
Updated!